src/traces/bar/set_positions.js

```diff
@@ -179,6 +179,24 @@ function setGroupPositionsInStackOrRelativeMode(gd, pa, sa, calcTraces) {
 
     // set bar bases and sizes, and update size axis
     stackBars(gd, sa, sieve);
+
+    // flag the outmost bar (for text display purposes)
+    for(var i = 0; i < calcTraces.length; i++) {
```

Comment:
Last night, you made me hesitate! But after further thought, here's the correct reasoning.

Bars with `textposition: 'none'` are still visible (only the bar text is hidden) and thus they can be the outmost bar.